app-modules/service-management/src/Observers/ServiceRequestObserver.php

```diff
@@ -80,6 +81,12 @@ public function saving(ServiceRequest $serviceRequest): void
         if ($serviceRequest->wasChanged('status_id')) {
             $serviceRequest->status_updated_at = now();
         }
+
+        if ($serviceRequest->status->classification === SystemServiceRequestClassification::Closed) {
+            if ($serviceRequest->time_to_resolution === null) {
+                UpdateTTR::dispatch($serviceRequest);
+            }
+        }
```

Comment:
Please move this within the `if` statement above, and change the `if` check to `isDirty` not `wasChanged`

Reply:
@Orrison Done